<a href="https://colab.research.google.com/github/2241031c/Night-Lights-Dissertation/blob/main/NL_10_7_24.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#This script converts between radiance, Luminance and NELD values to be used to cross reference with IDA accreditation

In [1]:
#Import library
import math

In [4]:
def convert_mcd_m2_to_nw_sr_cm2_corrected(luminance_mcd_m2):
    # Convert mcd/m² to cd/m²
    luminance_cd_m2 = luminance_mcd_m2 / 1000

    # Conversion factor from cd/m² to W/sr/m² using luminous efficacy
    luminous_efficacy = 683  # lm/W for photopic vision

    # Convert cd/m² to W/sr/m²
    radiance_w_sr_m2 = luminance_cd_m2 / luminous_efficacy

    # Convert W/sr/m² to nW/sr/cm²
    radiance_nw_sr_cm2 = radiance_w_sr_m2 * 1e7

    return radiance_nw_sr_cm2

# Example usage
luminance_mcd_m2 = 0.4
radiance_nw_sr_cm2_corrected = convert_mcd_m2_to_nw_sr_cm2_corrected(luminance_mcd_m2)
radiance_nw_sr_cm2_corrected


5.856515373352855

Luminance: 0.4 mcd/m² is equivalent to Radiance: 5.857 nW/sr/cm²